src/coreclr/src/jit/compiler.cpp

```diff
@@ -5251,14 +5251,20 @@ int Compiler::compCompile(CORINFO_METHOD_HANDLE methodHnd,
 
     virtualStubParamInfo = new (this, CMK_Unknown) VirtualStubParamInfo(IsTargetAbi(CORINFO_CORERT_ABI));
 
+    // compMatchedVM is true if both CPU/ABI and OS are matching the execution engine requirements
+    // 
     // Do we have a matched VM? Or are we "abusing" the VM to help us do JIT work (such as using an x86 native VM
     // with an ARM-targeting "altjit").
+    // Match CPU/ABI for compMatchedVM
     info.compMatchedVM = IMAGE_FILE_MACHINE_TARGET == info.compCompHnd->getExpectedTargetArchitecture();
 
-#if (defined(_TARGET_UNIX_) && !defined(_HOST_UNIX_)) || (!defined(_TARGET_UNIX_) && defined(_HOST_UNIX_))
-    // The host and target platforms don't match. This info isn't handled by the existing
-    // getExpectedTargetArchitecture() JIT-EE interface method.
-    info.compMatchedVM = false;
+    // Match OS for compMatchedVM
+    CORINFO_EE_INFO eeInfo;
+    info.compCompHnd->getEEInfo(&eeInfo);
+#ifdef _TARGET_UNIX
```

Comment:
```suggestion
#ifdef _TARGET_UNIX_
```

I think this is causing a lot of tests to fail

Reply:
A quick glance at the reported failures was for "send tests to helix" but couldn't find any errors with more explanation, I will try to run the tests locally on my machine on both Linux and Windows.

Reply:
There is a problem in Azure DevOps right now that the test results are not visible from anonymous sessions. I guess it is why you are not able to find more detailed errors.

I am pretty sure that this typo in the ifdef is causing the test failures.

Reply:
Oops, I completely missed the trailing `_`  ok, it is hopefully only this problem